src/pytest_run_parallel/utils.py

```diff
@@ -41,26 +44,55 @@ def __init__(self, fn):
         super().__init__()
 
     def visit_Call(self, node):
+        if self.catches_warns:
+            return
+
         if isinstance(node.func, ast.Attribute):
             if isinstance(node.func.value, ast.Name):
                 real_mod = node.func.value.id
                 if real_mod in self.modules_aliases:
                     real_mod = self.modules_aliases[real_mod]
                 if (real_mod, node.func.attr) in self.blacklist:
                     self.catches_warns = True
+                elif self.level < 2:
+                    if node.func.value.id in getattr(self.fn, "__globals__", {}):
+                        mod = self.fn.__globals__[node.func.value.id]
+                        child_fn = getattr(mod, node.func.attr, None)
+                        if child_fn is not None:
+                            self.catches_warns = identify_warnings_handling(
+                                child_fn, self.skip_set, self.level + 1
+                            )
         elif isinstance(node.func, ast.Name):
+            recurse = True
             if node.func.id in self.func_aliases:
                 if self.func_aliases[node.func.id] in self.blacklist:
                     self.catches_warns = True
-
-
-def identify_warnings_handling(fn):
+                    recurse = False
+            if recurse and self.level < 2:
+                if node.func.id in getattr(self.fn, "__globals__", {}):
+                    child_fn = self.fn.__globals__[node.func.id]
+                    self.catches_warns = identify_warnings_handling(
+                        child_fn, self.skip_set, self.level + 1
+                    )
+
+    def visit_Assign(self, node):
+        if self.catches_warns:
+            return
+
+        if len(node.targets) == 1:
+            name_node = node.targets[0]
+            value_node = node.value
+            if getattr(name_node, "id", None) == "__thread_unsafe__":
```

Comment:
```suggestion
            if hasattr(name_node, "id", None) == "__thread_unsafe__":
```

That way the default is the same as if the attribute isn't defined at all and it doesn't matter what the value is.

Reply:
I think `hasattr` does not allow for 3 arguments, whereas `getattr` does.

Reply:
Ah yeah, sorry. Not literally that diff!

Reply:
Unless you refer to marking the test as thread-unsafe by just defining `__thread_unsafe__`

Reply:
Yeah, that’s exactly right. I’m saying you should only need to define the attribute and it can have any value.

If you want to keep the boolean flag that’s fine, but I think it should be spelled `__thread_safe__ = False` so the default is True. People have an easier time thinking about boolean flags that default to True.